packages/lit-html/src/lit-html.ts

```diff
@@ -399,14 +395,13 @@ class Template {
         }
       }
 
-      // console.assert(!(attrNameEnd !== -1 && regex === textRegex));
-      if (attrNameEnd !== -1) {
-        attrNames.push(attrName!);
-        html +=
-          s.slice(0, attrNameEnd) + '$lit$' + s.slice(attrNameEnd) + marker;
-      } else {
-        html += regex === textRegex ? s + nodeMarker : s + marker;
-      }
+      html +=
+        regex === textRegex
```

Comment:
As discussed, you might have a hanging `attrNameEnd` with other states (comment or raw text element), need to not emit the attribute in those states too.  So need to add tests for those, and either make sure `attrNameEnd` is always cleared when exiting attribute/tag position, or change the test here.

Reply:
That needed to be cleaned up to handle raw text elements. Added tests.

Reply:
Suggest adding a comment explaining what's happening here with some examples.

Reply:
If changing the ordering of the conditions is important, call that out in the comment.